### PR TITLE
Change topic_name variable to output_topic_name

### LINKED ISSUE #28: Change `topic_name` variable to `output_topic_name`
Currently, in the config.yaml we define the default topic names like this:
```yaml
topic_name_config:
  default_error_topic_name: "${pipeline_name}-${component_name}-dead-letter-topic"
  default_output_topic_name: "${pipeline_name}-${component_name}-topic"
```
we use the `${topic_name}` in the `pipeline.yaml`. This internal variable is replaced by the `default_output_topic_name`. The naming of this variable is misleading and should be changed to `${output_topic_name}`.
Like the `${error_topic_name}` variable.

### PR DESCRIPTION
**Pull request by MichaelKora**
Fixes #28

### CODE REVIEW

**Review by raminqaf — COMMENTED** [review]
Could you please add a test to the `test_pipeline.py` that uses these variables in the `pipeline.yaml`?

File: tests/pipeline/test_pipeline.py [hunk 1]
```diff
@@ -208,3 +208,63 @@ def test_with_env_defaults(self, tmpdir, snapshot):
...
+            nameOverride = app_details["nameOverride"]
+            outup_topic = app_details["streams"]["outputTopic"]
+            error_topic = app_details["streams"]["errorTopic"]
+
+            assert outup_topic == nameOverride
+            assert error_topic == nameOverride + "-error"
+
+        snapshot.assert_match(enriched_pipeline, "test-pipeline")
+
+    def with_customed_config(self, tmpdir, snapshot):
```
Comment (disrupted) [inline tests/pipeline/test_pipeline.py:243]: either `custom_config` or `customized_config`

**Review by disrupted — CHANGES_REQUESTED** [review]

File: tests/pipeline/test_pipeline.py [hunk 2]
```diff
@@ -208,3 +208,34 @@ def test_with_env_defaults(self, tmpdir, snapshot):
...
+                output_file_path,
+                "--defaults",
+                str(RESOURCE_PATH / "no-topics-defaults"),
+                "--config",
+                str(RESOURCE_PATH / "custom-config/config.yaml"),
+            ],
+        )
+        assert result.exit_code == 0, result.exception
+
+        enriched_pipeline = load_yaml_file(Path(output_file_path))
```
Comment (disrupted) [inline tests/pipeline/test_pipeline.py:232]: please add
```suggestion
        enriched_pipeline = load_yaml_file(Path(output_file_path))
        assert isinstance(enriched_pipeline, dict)
```
for the type checker

**Review by disrupted — CHANGES_REQUESTED** [review]

File: tests/pipeline/test_pipeline.py [hunk 3]
```diff
@@ -208,3 +208,35 @@ def test_with_env_defaults(self, tmpdir, snapshot):
...
+        assert isinstance(enriched_pipeline, dict)
+        for app_details in enriched_pipeline["components"]:
+            output_topic = app_details["app"]["streams"]["outputTopic"]
+            assert output_topic == "random-topic"
+            app_type = app_details["type"]
+            if app_type == "streams-app":
+                error_topic = app_details["app"]["streams"]["errorTopic"]
+                assert error_topic == "random-error"
+
+        snapshot.assert_match(enriched_pipeline, "test-pipeline")
```
Comment (disrupted) [inline tests/pipeline/test_pipeline.py:242]: please add the new snapshot to this PR

**Review by disrupted — CHANGES_REQUESTED** [review]

**Pull request merged**